### ISSUE TITLE
Ignoring filtered volumes when wildcards are used

### ISSUE THREAD

**Issue by Massimo-B** [digint/btrbk#311]
I have multiple remote volumes like
```
volume /mnt/btrfs-top-lvl
    group all,cron,local
    #...
volume ssh://serverA/mnt/btrfs-top-lvl
    group all,cron,ssh,serverA
    #...
volume ssh://serverB/mnt/btrfs-top-lvl
    group all,cron,ssh,serverB
    #...
```
When filtering for serverA and serverB is not available I get errors like this:
```
# btrbk --progress -v run serverA
btrbk command line client, version 0.29.0  (Tue Dec 10 08:42:49 2019)
Using configuration: /etc/btrbk/btrbk.conf
Expanding wildcards: /mnt/btrfs-top-lvl/volumes.root/*
WARNING: No subvolumes found matching: /mnt/btrfs-top-lvl/volumes.root/*
Expanding wildcards: serverA:/mnt/btrfs-top-lvl/@root/*
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/boot
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/tmp
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/usr.portage
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/usr.src
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/var.cache
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/var.db.repos
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/var.lib.layman
Found source subvolume: serverA:/mnt/btrfs-top-lvl/@root/var.tmp
ERROR: Command execution failed (exitcode=255): `ssh -i /root/.ssh/id_rsa root@serverB 'readlink -v -e /mnt/btrfs-top-lvl'`
ERROR: ... sh: ssh -i /root/.ssh/id_rsa root@serverB 'readlink -v -e /mnt/btrfs-top-lvl'
ERROR: ... ssh: Could not resolve hostname serverB: Temporary failure in name resolution
WARNING: Skipping volume "serverB:/mnt/btrfs-top-lvl": Failed to fetch subvolume detail
WARNING: ... sh: ssh -i /root/.ssh/id_rsa root@serverB 'readlink -v -e /mnt/btrfs-top-lvl'
WARNING: ... ssh: Could not resolve hostname serverB: Temporary failure in name resolution

ERROR: process died unexpectedly (btrbk v0.29.0)
Please contact the author: Axel Burri <axel@tty0.ch>

Stack Trace:
----------------------------------------
Use of uninitialized value in join or string at /usr/sbin/btrbk line 3030.
 at /usr/sbin/btrbk line 307.
	main::__ANON__("Use of uninitialized value in join or string at /usr/sbin/btr"...) called at /usr/sbin/btrbk line 3030
	main::vinfo_match(ARRAY(0x555945b17278), HASH(0x555945cc7b40), "flag_matched", "_matched") called at /usr/sbin/btrbk line 5722
```
btrbk should not fail for other volumes not available. I would expect that other filtered volumes should be completely ignored without any warning.

Then the warning about  /mnt/btrfs-top-lvl/volumes.root/* should not happen as the local volume should be filtered with that command as well.

**Comment by Massimo-B** [comment]
More serious is the issue beside ignored filters:
When I have some volume ssh:// as part of my main btrbk configuration and some host is not available, the whole btrbk is failing like that.
At least for targets that are not available, like my usb-drive that is not always connected, those are just skipped. Please do the same for volumes. I don't see a benefit from an option marking those as optional or required.

**Comment by digint** [comment]
fixed in: 37b30caeb2d7913fc0b7ee4f7bdb7c2e550e8b85 btrbk: bugfix: vinfo_match: skip match against invalid paths

Skipping volumes / subvolumes should work as expected again.

**Comment by digint** [comment]
Fix included in [btrbk-v0.29.1](/digint/btrbk/releases/tag/v0.29.1).
